graphql_compiler/compiler/metadata.py

```diff
@@ -132,6 +151,18 @@ def get_location_info(self, location):
                                  u'{}'.format(location))
         return location_info
 
+    def record_explain_info(self, location, explain_info):
+        """Record estimation information about the location."""
+        if isinstance(location, FoldScopeLocation):
+            # NOTE(gurer): ignore filters inside the fold for now
+            return
+        record_location = location.at_vertex()
+        self._explain_infos[record_location].append(explain_info)
```

Comment:
It might be useful to draw a distinction here between explain infos we only expect 1 of (recurse), and explain infos we expect multiple of (filters). This could get weird if we mistakenly append 2 `RecurseInfo` tuples. 

This might end up being more ergonomic. Right now to ask a question about a `RecurseInfo` at the current location, we have to get all explain infos and then take the first things that's a `RecurseInfo`, vs storing this separately to enforce the uniqueness property.

Reply:
I have separated them. It made more sense. You can directly get RecurseInfo as you mentioned. Recording it twice gives an assertion. You can loop over filters without checking.